File: server/src/main/java/org/example/filesharing/entities/dtos/MetadataDTO.java
package org.example.filesharing.entities.dtos;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MetadataDTO {
    private String fileId;
    private String fileName;
    private String mimeType;
    private Double fileSize;
    private String compressionAlgo;
    private int ownerId;
    private LocalDateTime creationTimestamp;
    private int timeToLive;
    private List<ChunkRequest> chunkList;
}

File: client/src/dto/MetadataDto.ts
export interface MetadataDto {
    fileId: string;
    fileName: string;
    objectName: string;
    mimeType: string;
    fileSize: number;
    compressionAlgo: string;
    ownerId: string;
    creationTimestamp: Date;
    timeToLive: number;
}